pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1319,13 +1319,29 @@ func (d *awsDisk) describeVolume() (*ec2.Volume, error) {
 // waitForAttachmentStatus polls until the attachment status is the expected value
 // On success, it returns the last attachment state.
 func (d *awsDisk) waitForAttachmentStatus(status string) (*ec2.VolumeAttachment, error) {
-	attempt := 0
-	maxAttempts := 60
+	// We wait up to 30 minutes for the attachment to complete.
+	// This mirrors the GCE timeout.
+	timeoutAt := time.Now().UTC().Add(30 * time.Minute).Unix()
+
+	// Because of rate limiting, we often see errors from describeVolume
+	// So we tolerate a limited number of failures.
+	// But once we see more than tolerateDescribeErrors in a row, we return the error
+	describeErrorCount := 0
+	tolerateDescribeErrors := 5
```

Comment:
Also put this number into constant? Also is 5 enough? I think to be safe, it is ok to retry a little bit more? It is more dangerous if controller thinks it is still attached but a detaching is still pending or eventually happens successfully.